src/System.Memory/src/System/Buffers/ReadOnlySequence_TryGet.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Buffers
+{
+    public readonly partial struct ReadOnlySequence<T>
+    {
+        internal bool TryGetMemoryList(out IMemoryList<T> startSegment, out int startIndex, out IMemoryList<T> endSegment, out int endIndex)
```

Comment:
Why are these new TryGet methods in a new file? Can we keep all these methods in ReadOnlySequence_helpers.cs?